lib/logger/lib/logger.ex

```diff
@@ -442,7 +445,10 @@ defmodule Logger do
   def remove_backend(backend, opts \\ []) do
     _ = if opts[:flush], do: flush()
     Logger.Config.remove_backend(backend)
-    Logger.Watcher.unwatch(Logger, Logger.Config.translate_backend(backend))
+    case Logger.Watcher.unwatch(Logger, Logger.Config.translate_backend(backend)) do
+      {:error, :module_not_found} -> {:error, :not_found}
```

Comment:
This value shouldn't be possible now because switched to using `Supervisor.terminate_child/2`, which returns `{:error, :not_found}`.